projects/storefrontlib/src/lib/cms/services/occ-cms.service.ts

```diff
@@ -48,7 +54,7 @@ export class OccCmsService {
     id: string,
     pageContext: PageContext,
     fields?: string
-  ): Observable<any> {
+  ): Observable<Component> {
```

Comment:
use `<T extends Component>`

Reply:
This causes a syntax error

Reply:
This causes a syntax error

Reply:
loadComponent<T extends Component>(
    id: string,
    pageContext: PageContext,
    fields?: string
  ): Observable<T> {
causes errors
Type Object is not assignable to type T

Reply:
Also, doing just `<T extends Component>` (to replace `Component`) yields error <cannot find name T>

Reply:
`
loadComponent<T extends CmsComponent>(
    id: string,
    pageContext: PageContext,
    fields?: string
  ): Observable<T> {
    return this.http
      .get<T>(this.getBaseEndPoint() + `/components/${id}`, {
        headers: this.headers,
        params: new HttpParams({
          fromString: this.getRequestParams(pageContext, fields)
        })
      })
      .pipe(catchError((error: any) => throwError(error.json())));
  }`

I will push my codes.